dashboard/app/models/concerns/localizable.rb

```diff
@@ -0,0 +1,172 @@
+    end
+
+    # Class method to define localizable attributes
+    def localizable_attributes=(*attrs)
+      @localizable_attributes = attrs.flatten.map(&:to_sym)
+
+      # Generate localized_* methods for each configured attribute
+      @localizable_attributes.each do |attr|
+        define_method("localized_#{attr}") do |locale_code = I18n.locale, key_override: nil|
+          localize_property(attr, locale_code: locale_code, key_override: key_override)
+        end
+
+        # Also create a method that accepts options hash (for backward compatibility)
+        define_method("localized_#{attr}_with_options") do |options = {}|
+          locale_code = options[:locale] || options[:locale_code] || I18n.locale
+          key_override = options[:key]
+          localize_property(attr, locale_code: locale_code, key_override: key_override)
+        end
+      end
+    end
+
+    # Helper to check if a class has any localizable attributes
+    def has_localizable_attributes?
+      localizable_attributes.any?
+    end
+  end
+
+  # Main localization method
+  def localize_property(property_name, locale_code: I18n.locale, key_override: nil)
+    property_name = property_name.to_sym
+    cache_key = "#{property_name}_#{locale_code}"
+
+    # Validate that the property is configured as localizable
+    unless self.class.localizable_attributes.include?(property_name)
+      CDO.log.warn("Attempting to localize non-configured attribute #{property_name} on #{self.class.name}")
+      return public_send(property_name) if respond_to?(property_name)
+      return nil
+    end
+
+    # Get the fallback value (original property value)
+    fallback_value = get_fallback_value(property_name)
+    # Return nil immediately if the fallback is nil (don't try to localize nil values)
+    if fallback_value.nil?
+      @localization_cache[cache_key] = nil if @localization_cache
+      return nil
+    end
+
+    # Return fallback if we're using the default locale
+    return fallback_value if locale_code.to_s == I18n.default_locale.to_s
+
+    # Return memoize translation if it exists
+    return @localization_cache[cache_key] if @localization_cache&.key?(cache_key)
+
+    # Determine lookup key: use override if provided, otherwise use object.key
+    lookup_key = key_override
+    lookup_key = key if lookup_key.blank? && respond_to?(:key)
+
+    result = if lookup_key.blank?
+               fallback_value
+             else
+               perform_i18n_lookup(property_name, fallback_value, locale_code, lookup_key)
+             end
+
+    # Cache the translation
+    @localization_cache[cache_key] = result if @localization_cache
+    result
+  end
+
+  # Check if this instance uses localization
+  def uses_localization?
+    self.class.has_localizable_attributes?
+  end
+
+  # Clear localization cache (useful after updates)
+  def clear_localization_cache!
+    @localization_cache&.clear
+  end
+
+  private def get_fallback_value(property_name)
```

Comment:
https://github.com/rubocop/ruby-style-guide?tab=readme-ov-file#accessormutator-method-names
```suggestion
  private def fallback_value_for(property_name)
```